sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_models.py

```diff
@@ -0,0 +1,100 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=too-few-public-methods, too-many-instance-attributes
+# pylint: disable=super-init-not-called, too-many-lines
+from azure.core.async_paging import AsyncPageIterator
+
+from .._deserialize import return_headers_and_deserialized_path_list, process_storage_error
+from .._generated.models import StorageErrorException
+from .._models import PathPropertiesPaged as PathPropertiesPagedBase
+
+from .._models import FileSystemProperties, ContainerPropertiesPaged
+
+
+class FileSystemPropertiesPaged(ContainerPropertiesPaged):
```

Comment:
Should this by using the AsyncPageIterator?

Reply:
good catch!! I was trying to merge the import... this merge broke the test haha...

Reply:
ContainerPropertiesPaged inherits AsyncPageIterator, so FileSystemPropertiesPaged also inherits AsyncPageIterator.
File System has the same concept as Container. We don't want to return ContainerPropertiesPaged because we don't want to confuse the customers.